sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+
+/**
+ * @brief remove a timer
+ *
+ * @note this function runs in O(n) with n being the number of active timers
+ *
+ * @param[in] timer ptr to timer structure that will be removed
+ *
+ * @return 1 on success
+ * @return 0 when timer was not active
+ */
+int xtimer_remove(xtimer_t *timer);
+
+/**
+ * @brief receive a message blocking but with timeout
+ *
+ * @param[out]   m           pointer to a msg_t which will be filled in case of no timeout
+ * @param[in]    us          timeout in microseconds relative
+ *
+ * @return       < 0 on error, other value otherwise
+ */
+int xtimer_msg_receive_timeout(msg_t *m, uint32_t us);
+
+/**
+ * @brief receive a message blocking but with timeout, 64bit version
+ *
+ * @param[out]   m           pointer to a msg_t which will be filled in case of no timeout
+ * @param[in]    us          timeout in microseconds relative
+ *
+ * @return       < 0 on error, other value otherwise
+ */
+int xtimer_msg_receive_timeout64(msg_t *m, uint64_t us);
+
+/**
+ * @brief xtimer backoff value
+ *
+ * All timers that are less than XTIMER_BACKOFF microseconds in the future will
+ * just spin.
+ *
+ * This is supposed to be defined per-device in e.g., periph_conf.h.
+ */
+#ifndef XTIMER_BACKOFF
+#define XTIMER_BACKOFF 30
+#endif
+
+/**
+ * @brief xtimer overhead value
+ *
+ * This value specifies the time a timer will be late if uncorrected, e.g.,
+ * the system-specific xtimer execution time from timer ISR to executing
+ * a timer's callback's first instruction.
+ *
+ * E.g., with XTIMER_OVERHEAD == 0
+ * start=xtimer_now();
+ * xtimer_set(&timer, X);
+ * (in callback:)
+ * overhead=xtimer_now()-start-X;
+ *
+ * xtimer automatically substracts XTIMER_OVERHEAD from a timer's target time,
+ * but when the timer triggers, xtimer will spin-lock until a timer's target
+ * time is reached, so timers will never trigger early.
+ *
+ * This is supposed to be defined per-device in e.g., periph_conf.h.
+ */
+#ifndef XTIMER_OVERHEAD
+#define XTIMER_OVERHEAD 20
+#endif
+
+#ifndef XTIMER_ISR_BACKOFF
+/**
+ * @brief   xtimer isr backoff time
+ *
+ * When scheduling the next isr, if it is less than the backoff time
+ * in the future, just spin.
+ */
+#define XTIMER_ISR_BACKOFF 20
+#endif
+
+/**
+ * @brief set xtimer default timer configuration
+ * @{
+ */
+#ifndef XTIMER
+#define XTIMER (0)
+#define XTIMER_CHAN (0)
+
+#if TIMER_0_MAX_VALUE == 0xffffff
+#define XTIMER_MASK 0xff000000
+#elif TIMER_0_MAX_VALUE == 0xffff
+#define XTIMER_MASK 0xffff0000
+#endif
+
+#endif
+/**
+ * @}
+ */
+
+#ifndef XTIMER_MASK
+/**
+ * @brief xtimer timer mask
+ *
+ * This value specifies the mask relative to 0xffffffff that the used timer
+ * counts to, e.g., 0xffffffff & ~TIMER_MAXVALUE.
+ *
+ * For a 16bit timer, the mask would be 0xFFFF0000, for a 24bit timer, the mask
+ * would be 0xFF000000. Don't set this for 32bit timers.
+ *
+ * This is supposed to be defined per-device in e.g., periph_conf.h.
+ */
+#define XTIMER_MASK 0
```

Comment:
Warning is a GCC feature: 

```C
#include <stdio.h>
#warning "Test"
int main(int argc, char **argv)
{
    return 0;
}
```

```bash
CFLAGS="-Wall -pedantic -Wextra" make main
```

```
cc -Wall -pedantic -Wextra    main.c   -o main
main.c:17:2: warning: #warning is a GCC extension [enabled by default]
 #warning "Test"
  ^
main.c:17:2: warning: #warning "Test" [-Wcpp]
main.c: In function ‘main’:
main.c:19:14: warning: unused parameter ‘argc’ [-Wunused-parameter]
 int main(int argc, char **argv)
              ^
main.c:19:27: warning: unused parameter ‘argv’ [-Wunused-parameter]
 int main(int argc, char **argv)
                           ^
```

My clang version is building, but I'm not sure if that is guaranteed for all versions:

```bash
CC=clang CFLAGS="-Wall -pedantic -Wextra" make main
```

```
clang -Wall -pedantic -Wextra    main.c   -o main
main.c:17:2: warning: #warning is a language extension [-Wpedantic]
#warning "Test"
 ^
main.c:17:2: warning: "Test" [-W#warnings]
main.c:19:14: warning: unused parameter 'argc' [-Wunused-parameter]
int main(int argc, char **argv)
             ^
main.c:19:27: warning: unused parameter 'argv' [-Wunused-parameter]
int main(int argc, char **argv)
                          ^
4 warnings generated.
```